File: JAVA/Product.java
class Product {     //class orangtua, yang akan mewarisi atributnya ke anak - cucu
    int idProduct;  //deklarasi atribut
    String name;
    String brand;
    String price;

    Product (){ //konstructor kosong
    }

    Product (int idProduct, String name, String brand, String price){   //konstruktor yang ada parameter
        this.idProduct = idProduct;
        this.name = name;
        this.brand = brand;
        this.price = price;
    }

    //get/set atribut2
    public int getIdProduct() {
        return idProduct;
    }

    public void setIdProduct(int idProduct) {
        this.idProduct = idProduct;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getBrand() {
        return brand;
    }

    public void setBrand(String brand) {
        this.brand = brand;
    }

    public String getPrice() {
        return price;
    }

    public void setPrice(String price) {
        this.price = price;
    }

}

File: CPP/PROGRAM/Product.cpp
//product -> clothing ->shirt
#include <iostream>
#include <string>   //desklarasi library yang ingin dipakai

//using standard namespace
using namespace std;
 
class Product{
    private:
        //deklarasi object yang ada di dalam class product
        string name;
        string idProduct;
        string brand;
        string price;

    public:
        //konstruktor
        Product(){
        }

        //getter dan setter
        void set_name(string name){
            this->name = name;
        }

        string get_name(){
            return this->name;
        }

        void set_idProduct(string idProduct){
            this->idProduct = idProduct;
        }

        string get_idProduct(){
            return this->idProduct;
        }

        void set_brand(string brand){
            this->brand = brand;
        }

        string get_brand(){
            return this->brand;
        }

        void set_price(string price){
            this->price = price;
        }

        string get_price(){
            return this->price;
        }

        ~Product(){
        }
};
